Aplib.Core/Collections/ExposedQueue.cs

```diff
@@ -45,30 +46,30 @@ public ExposedQueue(int size)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="ExposedQueue{T}"/> class
+        /// Initializes a new instance of the <see cref="ExposedQueue{T}" /> class
         /// with an array to use as basis for the queue.
-        /// By default, assumes the array is filled. 
+        /// By default, assumes the array is filled.
         /// </summary>
         /// <param name="array">An array to use as the circular array.</param>
-        /// <param name="count">The number of actual elements in the array.</param>
+        /// <param name="maxCount">The number of actual elements in the array.</param>
         /// <remarks>
         /// The MaxCount of the queue will be set to the length of the array.
         /// If the array is not fully filled, the Count should be specified.
         /// </remarks>
-        public ExposedQueue(T[] array, int count)
+        public ExposedQueue(T[] array, int maxCount)
         {
-            if (count > array.Length)
-                throw new ArgumentOutOfRangeException(nameof(count), "Count cannot exceed the length of the array.");
-            if (count < 0)
-                throw new ArgumentOutOfRangeException(nameof(count), "Count cannot be negative.");
+            if (maxCount < 0) throw new ArgumentOutOfRangeException(nameof(maxCount), "Count cannot be negative.");
 
-            MaxCount = array.Length;
-            _array = array;
-            _head = MaxCount - 1;
-            Count = count;
+            MaxCount = maxCount;
+            Count = Math.Min(array.Length, maxCount);
```

Comment:
It seems now the constructor always assumes the array is full, there is no way to specify that you are passing an array where only x slots are filled.
How do we want this to handle empty arrays? (Open question, I don't know the best way)

Reply:
I'd say we don't handle them. If I'm passing an empty array, I'd expect the resulting ExposedQueue to have those empty spots as well. It doesn't break the code, so I'd leave it like this.